containers/hawkular-metrics-index-truncator/src/main/java/org/hawkular/openshift/truncator/Truncator.java

```diff
@@ -0,0 +1,144 @@
+        metricsRC = System.getProperty("hawkular.metrics.rc-name", METRICS_RC_NAME_DEFAULT);
+        heapsterRC = System.getProperty("hawkular.metrics.heapster.rc-name", HEAPSTER_RC_NAME_DEFAULT);
+        cassandraPodPrefix = System.getProperty("hawkular.metrics.cassandra.pod-prefix", CASSANDRA_POD_PREFIX_DEFAULT);
+        partitionThreshold = Integer.getInteger("hawkular.metrics.partition-threshold", PARTITION_THRESHOLD_DEFAULT);
+        useSSL = Boolean.getBoolean("hawkular.metrics.cassandra.use-ssl");
+        forceTruncation = Boolean.getBoolean("hawkular.metrics.force-truncation");
+        keyspace = System.getProperty("hawkular.metrics.cassandra.keyspace", "hawkular_metrics");
+        namespace = System.getProperty("hawkular.metrics.namespace", "openshift-infra");
+        cassandraConnectionMaxDelay = Long.getLong("hawkular.metrics.cassandra.connection.max-delay", 30);
+        cassandraConnectionMaxRetries = Integer.getInteger("hawkular.metrics.cassandra.connection.max-retries", 5);
+        truncatorMaxRetries = Integer.getInteger("hawkular.metrics.truncator.max-retries", 5);
+        truncatorMaxDelay = Long.getLong("hawkular.metrics.truncator.max-retries", 5*1000);
+    }
+
+    private String getToken() throws IOException {
+        return new String(Files.readAllBytes(Paths.get(TOKEN_FILE_LOCATION)));
+    }
+
+    public void truncate() throws IOException, InterruptedException, RuntimeException {
+
+        String token = this.getToken();
+        openshiftClient = new OpenshiftClientWrapper(this.namespace, this.KUBERNETES_MASTER_URL, token);
+
+        Pod[] cassandraPods = openshiftClient.getPodsWithPrefix(this.cassandraPodPrefix);
+        ArrayList<String> cassandraHosts = new ArrayList<>();
+        Arrays.stream(cassandraPods).forEach((pod) -> {
+            cassandraHosts.add(pod.getIP());
+        });
+
+        if (cassandraPods.length > 0 || this.forceTruncation) {
+
+            if (openshiftClient.hasLargePartition(this.partitionThreshold)) {
+                IndexTruncator truncator = new IndexTruncator(cassandraHosts, this.keyspace, this.useSSL);
+                truncator.setCassandraConnectionMaxDelay(cassandraConnectionMaxDelay);
+                truncator.setCassandraConnectionMaxRetries(cassandraConnectionMaxRetries);
+                // Get current replicas
+                int metricsReplicas = openshiftClient.getReplicationControllerScale(this.metricsRC);
+                int heapsterReplicas = openshiftClient.getReplicationControllerScale(this.heapsterRC);
+
+                // Scale to zero
+                openshiftClient.scaleReplicaitonController(this.heapsterRC, 0);
+                openshiftClient.scaleReplicaitonController(this.metricsRC, 0);
+
+                // Truncate
+                truncator.execute();
+
+                // Restore RC, this should rebuild indices.
+                openshiftClient.scaleReplicaitonController(this.heapsterRC, heapsterReplicas);
```

Comment:
the scaling back should be performed always even if `truncator.execute()` fails so we don't leave the cluster without hawkular and heapster